bgpd/bgp_io.c

```diff
@@ -231,8 +229,10 @@ static int bgp_process_reads(struct thread *thread)
 		 */
 		if (ringbuf_remain(ibw) >= pktsize) {
 			struct stream *pkt = stream_new(pktsize);
-			assert(ringbuf_get(ibw, pktbuf, pktsize) == pktsize);
-			stream_put(pkt, pktbuf, pktsize);
+			/* Copy the data to steam buffer */
+			assert(pkt != NULL);
```

Comment:
I don't think this could be NULL at all. If malloc fails, then we have more problems.